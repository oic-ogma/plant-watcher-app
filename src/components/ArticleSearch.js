import React, { Component } from 'react';
import { Button } from  'react-native-elements';
import { View, Keyboard } from 'react-native';
import { FormValidationMessage } from 'react-native-elements';
import { SearchBarComponent } from './common/SearchBar';
import { Actions } from 'react-native-router-flux';

export default class ArticleSearch extends Component {
  onSubmit() {
    const { textSearchArticle, plantName } = this.props;
    Keyboard.dismiss();
    textSearchArticle(plantName);
  }
  render() {
    const { searchArticlePlantNameChanged, plantName, error } = this.props;
    const { buttonStyle } = styles;

    return (
      <View style={{ marginTop: 140 }}>
        <SearchBarComponent
          onChangeText={searchArticlePlantNameChanged}
          placeholder='植物名'
          onSubmitEditing={this.onSubmit.bind(this)}
          value={plantName}
        />

        <FormValidationMessage>
          {error}
        </FormValidationMessage>

        <Button
          raised
          large
          icon={{ name: 'camera-alt' }}
          containerViewStyle={buttonStyle}
          title='カメラで撮影して検索'
          backgroundColor='green'
          onPress={Actions.imagesearch}
        />
      </View>
    );
  }
}

const styles = {
  buttonStyle: {
    marginTop: 50,
    marginBottom: 30
  }
};
