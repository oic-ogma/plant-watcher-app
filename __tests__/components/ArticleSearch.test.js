import React from 'react';
import ArticleSearch from '../../src/components/ArticleSearch';
import renderer from 'react-test-renderer';

describe('記事検索', () => {
  it('snapshot-普通', () => {
    const tree = renderer.create(<ArticleSearch />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
