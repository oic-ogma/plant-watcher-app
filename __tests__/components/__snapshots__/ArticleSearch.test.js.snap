// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`記事検索 snapshot-errorが出ている 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,0)",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "height": 650,
        "width": "100%",
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 650,
            "width": "100%",
          },
          undefined,
        ]
      }
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        Object {
          "fontSize": 50,
        },
        undefined,
        undefined,
        undefined,
        Object {},
        undefined,
        undefined,
        undefined,
        Object {
          "fontFamily": "Great Vibes",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "color": "rgb(255,255,255)",
          "marginBottom": "15%",
          "marginTop": "15%",
          "textAlign": "center",
        },
      ]
    }
  >
    Plant Watcher
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#393e42",
          "borderBottomColor": "#000",
          "borderBottomWidth": 1,
          "borderTopColor": "#000",
          "borderTopWidth": 1,
        },
        Object {
          "backgroundColor": "#e1e8ee",
          "borderBottomColor": "#e1e1e1",
          "borderTopColor": "#e1e1e1",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "borderBottomColor": "rgba(0,0,0,0)",
          "borderTopColor": "rgba(0,0,0,0)",
        },
      ]
    }
  >
    <TextInput
      onChangeText={undefined}
      onSubmitEditing={[Function]}
      placeholder="植物名"
      placeholderTextColor="#86939e"
      selectionColor="#86939e"
      style={
        Array [
          Object {
            "backgroundColor": "#303337",
            "borderRadius": 3,
            "color": "#86939e",
            "fontSize": 17.5,
            "height": 30,
            "margin": 8,
            "overflow": "hidden",
            "paddingLeft": 26,
            "paddingRight": 19,
          },
          Object {
            "backgroundColor": "#bdc6cf",
          },
          false,
          Object {
            "borderRadius": 15,
          },
          Object {
            "fontSize": 20,
            "height": 50,
            "marginLeft": "10%",
            "marginRight": "10%",
            "paddingLeft": "10%",
          },
          null,
          false,
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#86939e",
            "fontSize": 16,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
              "position": "absolute",
              "top": 15.5,
            },
            Object {
              "left": 16,
            },
            Object {
              "fontSize": 30,
              "marginLeft": "8%",
              "top": 17.5,
              "width": 30,
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#ff190c",
              "fontSize": 15,
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 5,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      ほげほげ
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "marginLeft": 15,
          "marginRight": 15,
        },
        Object {
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        Object {
          "marginBottom": 30,
          "marginTop": 10,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "green",
            },
            undefined,
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            Object {
              "height": "48%",
              "marginLeft": "21%",
              "width": "60%",
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 18,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                Object {
                  "fontSize": 23,
                },
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          カメラで検索
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`記事検索 snapshot-plantNameが入力されている 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,0)",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "height": 650,
        "width": "100%",
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 650,
            "width": "100%",
          },
          undefined,
        ]
      }
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        Object {
          "fontSize": 50,
        },
        undefined,
        undefined,
        undefined,
        Object {},
        undefined,
        undefined,
        undefined,
        Object {
          "fontFamily": "Great Vibes",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "color": "rgb(255,255,255)",
          "marginBottom": "15%",
          "marginTop": "15%",
          "textAlign": "center",
        },
      ]
    }
  >
    Plant Watcher
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#393e42",
          "borderBottomColor": "#000",
          "borderBottomWidth": 1,
          "borderTopColor": "#000",
          "borderTopWidth": 1,
        },
        Object {
          "backgroundColor": "#e1e8ee",
          "borderBottomColor": "#e1e1e1",
          "borderTopColor": "#e1e1e1",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "borderBottomColor": "rgba(0,0,0,0)",
          "borderTopColor": "rgba(0,0,0,0)",
        },
      ]
    }
  >
    <TextInput
      onChangeText={undefined}
      onSubmitEditing={[Function]}
      placeholder="植物名"
      placeholderTextColor="#86939e"
      selectionColor="#86939e"
      style={
        Array [
          Object {
            "backgroundColor": "#303337",
            "borderRadius": 3,
            "color": "#86939e",
            "fontSize": 17.5,
            "height": 30,
            "margin": 8,
            "overflow": "hidden",
            "paddingLeft": 26,
            "paddingRight": 19,
          },
          Object {
            "backgroundColor": "#bdc6cf",
          },
          false,
          Object {
            "borderRadius": 15,
          },
          Object {
            "fontSize": 20,
            "height": 50,
            "marginLeft": "10%",
            "marginRight": "10%",
            "paddingLeft": "10%",
          },
          false,
          Object {
            "paddingRight": 30,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value="ひまわり"
    />
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#86939e",
            "fontSize": 16,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
              "position": "absolute",
              "top": 15.5,
            },
            Object {
              "left": 16,
            },
            Object {
              "fontSize": 30,
              "marginLeft": "8%",
              "top": 17.5,
              "width": 30,
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#86939e",
            "fontSize": 16,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
              "position": "absolute",
              "top": 15.5,
            },
            Object {
              "right": 16,
            },
            Object {
              "fontSize": 30,
              "marginRight": "3%",
              "top": 17.5,
              "width": 30,
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#ff190c",
              "fontSize": 15,
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 5,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "marginLeft": 15,
          "marginRight": 15,
        },
        Object {
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        Object {
          "marginBottom": 30,
          "marginTop": 10,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "green",
            },
            undefined,
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            Object {
              "height": "48%",
              "marginLeft": "21%",
              "width": "60%",
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 18,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                Object {
                  "fontSize": 23,
                },
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          カメラで検索
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`記事検索 snapshot-普通 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,0)",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "height": 650,
        "width": "100%",
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 650,
            "width": "100%",
          },
          undefined,
        ]
      }
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        Object {
          "fontSize": 50,
        },
        undefined,
        undefined,
        undefined,
        Object {},
        undefined,
        undefined,
        undefined,
        Object {
          "fontFamily": "Great Vibes",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "color": "rgb(255,255,255)",
          "marginBottom": "15%",
          "marginTop": "15%",
          "textAlign": "center",
        },
      ]
    }
  >
    Plant Watcher
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#393e42",
          "borderBottomColor": "#000",
          "borderBottomWidth": 1,
          "borderTopColor": "#000",
          "borderTopWidth": 1,
        },
        Object {
          "backgroundColor": "#e1e8ee",
          "borderBottomColor": "#e1e1e1",
          "borderTopColor": "#e1e1e1",
        },
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "borderBottomColor": "rgba(0,0,0,0)",
          "borderTopColor": "rgba(0,0,0,0)",
        },
      ]
    }
  >
    <TextInput
      onChangeText={undefined}
      onSubmitEditing={[Function]}
      placeholder="植物名"
      placeholderTextColor="#86939e"
      selectionColor="#86939e"
      style={
        Array [
          Object {
            "backgroundColor": "#303337",
            "borderRadius": 3,
            "color": "#86939e",
            "fontSize": 17.5,
            "height": 30,
            "margin": 8,
            "overflow": "hidden",
            "paddingLeft": 26,
            "paddingRight": 19,
          },
          Object {
            "backgroundColor": "#bdc6cf",
          },
          false,
          Object {
            "borderRadius": 15,
          },
          Object {
            "fontSize": 20,
            "height": 50,
            "marginLeft": "10%",
            "marginRight": "10%",
            "paddingLeft": "10%",
          },
          null,
          false,
        ]
      }
      underlineColorAndroid="transparent"
      value={undefined}
    />
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#86939e",
            "fontSize": 16,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
              "position": "absolute",
              "top": 15.5,
            },
            Object {
              "left": 16,
            },
            Object {
              "fontSize": 30,
              "marginLeft": "8%",
              "top": 17.5,
              "width": 30,
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#ff190c",
              "fontSize": 15,
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 5,
            },
            undefined,
            undefined,
          ],
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "marginLeft": 15,
          "marginRight": 15,
        },
        Object {
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        Object {
          "marginBottom": 30,
          "marginTop": 10,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "green",
            },
            undefined,
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            Object {
              "height": "48%",
              "marginLeft": "21%",
              "width": "60%",
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 18,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                Object {
                  "fontSize": 23,
                },
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          カメラで検索
        </Text>
      </View>
    </View>
  </View>
</View>
`;
